README.md

```diff
@@ -1,83 +1,304 @@
+```
+
+Can you spot a problem in the code above?
+
+Instead of 10.5 ounces of coffee at 160°F, you are about to get 160 ounces of coffee at 10.5°F. (Yes that would be a very large block of ice... coffee.)
+
+Whoops, this is what it should have been:
+
+```swift
+let name = preferencesTuple.0
+let ounces = preferencesTuple.1
+let temperature = preferencesTuple.2
+```
+
+It would be great if instead of using .0, .1, and .2 you could use actual names. And you can, with named tuples!
+
+## Named Tuples
+
+In order to name the members of a tuple, all you have to do is add a variable name and a colon before the type:
+
+For example:
+```swift
+let namedTuple = (name: "Jim", age: 30)
 ```
 
-* Now, posing the question to the student... what happens if we iterate over the `favoriteDirectorAndTheirFilm` variable which is a dictionary of type [`String`: `String`] like we did with the `myFavoriteStarWarsFilms` array?
+The type of this tuple is `(name: String, age: Int)`.
 
-* Present some real world problem such as, “If you have a dictionary where the keys represent the seasons and the values represent your favorite color associated with that season, is there an easy way to print out a list that for each sentence and color that says “In the {season} my favorite color is {color}.”
-    - If you have a better example feel free to use it, but always aim for a real world example. And you can use that example throughout the rest of your reading and/or use other new examples. 
-    * The answer is tuples and this sets the stage for writing the content that covers your learning objectives above.--- You can iterate over the key-value pairs in a dictionary with a for-in loop. Each item in the dictionary is returned as a (key, value) tuple. Explain how you can decompose the tuple’s members into temporary constants as part of the iteration.
-    
+Now you can the access the values this way:
 ```swift
-let seasonFavColors = [
-    "Winter": "White",
-    "Spring": "Green",
-    "Summer": "Blue",
-    "Fall": "Orange"
+let name = namedTuple.name
+let age = namedTuple.age
+```
+
+Let's make the `preferencesForCustomer` method return a named tuple. You would update the method signature to the following:
+
+```swift
+func preferencesForCustomer(customerID: Int) -> (name: String, ounces: Double, temperature: Int)
+```
+
+Now you can access the values like this:
+
+```swift
+let preferencesTuple = preferencesForCustomer(customerID)
+let name = preferencesTuple.name
+let ounces = preferencesTuple.ounces
+let temperature = preferencesTuple.temperature
+```
+
+Which makes it a lot harder to make mistakes such as the following:
+```swift
+let ounces = preferencesTuple.temperature // Whoops, don't need 160 ounces of coffee
+```
+
+## Decomposing a Tuple
+
+You can also decompose a tuple into individual variables in a single line.
+
+```swift
+let preferencesTuple = preferencesForCustomer(customerID)
+let (name, ounces, temperature) = preferencesTuple
```

Comment:
I like this example, but I think we could be slightly more explicit here showing them code that goes through this. I think it's important for them to visualize and see the values of these various variables so that way they aren't stepping through the code in their code and getting lost (which might be an exaggeration).

```swift
let (name, ounces, temperature) = preferencesForCustomer(1234)
// Assuming the function call returns back ("Jolene", 9.0, 105)

print("\(name) likes his/her coffee being \(ounces)oz. & \(temperature)℉")
// prints "Jolene likes his/her coffee being 9.0oz. & 105℉"
```